catalog/forms.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
.cleaned_data có tác dụng gì em ?

Reply:
.cleaned_data để lấy ra dữ liệu của trường tương ứng mà người dùng đã nhập trên form và tự động chuyển dữ liệu này thành dạng tiêu chuẩn, chính xác (cleaned) để ta có thể sử dụng nó một cách an toàn trong các hàm phía sau.